node/impl/storminer.go

```diff
@@ -1387,3 +1387,43 @@ func (sm *StorageMinerAPI) ActorWithdrawBalance(ctx context.Context, amount abi.
 
 	return smsg.Cid(), nil
 }
+
+func (sm *StorageMinerAPI) BeneficiaryWithdrawBalance(ctx context.Context, minerAddress address.Address, amount abi.TokenAmount) (cid.Cid, error) {
```

Comment:
That's a lot of duplicated code -- let's extract all of this into a helper that takes a `useOwner` boolean param indicating who the sender should be?